include/commissioning/commissioner.h

```diff
@@ -50,24 +50,61 @@ extern "C" {
  * This function enables the Thread Commissioner role.
  *
  * @param[in]  aInstance         A pointer to an OpenThread instance.
- * @param[in]  aPSKd             A pointer to the PSKd.
- * @param[in]  aProvisioningUrl  A pointer to the Provisioning URL (may be NULL).
  *
- * @retval kThreadError_None         Successfully started the Commissioner role.
- * @retval kThreadError_InvalidArgs  @p aPSKd or @p aProvisioningUrl is invalid.
+ * @retval kThreadError_None     Successfully started the Commissioner role.
  *
  */
-ThreadError otCommissionerStart(otInstance *aInstance, const char *aPSKd, const char *aProvisioningUrl);
+ThreadError otCommissionerStart(otInstance *aInstance);
 
 /**
  * This function disables the Thread Commissioner role.
  *
- * @param[in]  aInstance  A pointer to an OpenThread instance.
+ * @param[in]  aInstance         A pointer to an OpenThread instance.
+ *
+ * @retval kThreadError_None     Successfully started the Commissioner role.
  *
  */
 ThreadError otCommissionerStop(otInstance *aInstance);
 
 /**
+ * This function adds a Joiner entry.
+ *
+ * @param[in]  aInstance        A pointer to an OpenThread instance.
+ * @param[in]  aExtAddress      A pointer to the Joiner's extended address.
+ * @param[in]  aPSKd            A pointer to the PSKd.
+ *
+ * @retval kThreadError_None         Successfully added the Joiner.
+ * @retval kThreadError_NoBufs       No buffers available to add the Joiner.
+ * @retval kThreadError_InvalidArgs  @p aExtAddress or @p aPSKd is invalid.
+ *
+ */
+ThreadError otCommissionerAddJoiner(otInstance *aInstance, const otExtAddress *aExtAddress, const char *aPSKd);
+
+/**
+ * This function removes a Joiner entry.
+ *
+ * @param[in]  aInstance          A pointer to an OpenThread instance.
+ * @param[in]  aExtAddress        A pointer to the Joiner's extended address.
+ *
+ * @retval kThreadError_None         Successfully added the Joiner.
+ * @retval kThreadError_NotFound     The Joiner specified by @p aExtAddress was not found.
+ * @retval kThreadError_InvalidArgs  @p aExtAddress is invalid.
+ *
+ */
+ThreadError otCommissionerRemoveJoiner(otInstance *aIntsance, const otExtAddress *aExtAddress);
+
+/**
+ * This function sets the Provisioning URL.
+ *
+ * @param[in]  aProvisioningUrl  A pointer to the Provisioning URL (may be NULL).
+ *
+ * @retval kThreadError_None         Successfully added the Joiner.
+ * @retval kThreadError_InvalidArgs  @p aProvisioningUrl is invalid.
+ *
+ */
+ThreadError otCommissionerSetProvisioningUrl(otInstance *aInstance, const char *aProvisioningUrl);
```

Comment:
Ok, sounds good.  I definitely agree more complex cases should focus on pushing processing to a more capable host processor or phone.